state/validation.go

```diff
@@ -99,12 +98,7 @@ func validateBlock(evidencePool EvidencePool, state State, block *types.Block) e
 	// NOTE: We can't actually verify it's the right proposer because we don't
 	// know what round the block was first proposed. So just check that it's
 	// a legit address and a known validator.
-	if len(block.ProposerAddress) != crypto.AddressSize {
-		return fmt.Errorf("expected ProposerAddress size %d, got %d",
-			crypto.AddressSize,
-			len(block.ProposerAddress),
-		)
-	}
+	// The length is checked in ValidateBasic above.
```

Comment:
I think it's probably better just to remove all commenting regarding checking the length of the proposer especially as this directly proceeds checking if the proposer is part of the current validator set. I feel like it might just be confusing